pkg/cmd/server/kubernetes/master/master_config.go

```diff
@@ -163,57 +170,30 @@ func BuildKubeAPIserverOptions(masterConfig configapi.MasterConfig) (*kapiserver
 }
 
 // BuildStorageFactory builds a storage factory based on server.Etcd.StorageConfig with overrides from masterConfig.
-// This storage factory is ONLY USED FOR kubernetes registries right now. Compare pkg/util/restoptions/configgetter.go.
-func BuildStorageFactory(masterConfig configapi.MasterConfig, server *kapiserveroptions.ServerRunOptions) (apiserverstorage.StorageFactory, error) {
+// This storage factory is used for kubernetes and origin registries. Compare pkg/util/restoptions/configgetter.go.
+func BuildStorageFactory(masterConfig configapi.MasterConfig, server *kapiserveroptions.ServerRunOptions, enforcedStorageVersions []schema.GroupVersionResource) (*apiserverstorage.DefaultStorageFactory, error) {
 	resourceEncodingConfig := apiserverstorage.NewDefaultResourceEncodingConfig(kapi.Registry)
-	resourceEncodingConfig.SetVersionEncoding(
-		kapi.GroupName,
-		schema.GroupVersion{Group: kapi.GroupName, Version: masterConfig.EtcdStorageConfig.KubernetesStorageVersion},
-		kapi.SchemeGroupVersion,
-	)
-
-	resourceEncodingConfig.SetVersionEncoding(
-		extensions.GroupName,
-		schema.GroupVersion{Group: extensions.GroupName, Version: "v1beta1"},
-		extensions.SchemeGroupVersion,
-	)
-
-	resourceEncodingConfig.SetVersionEncoding(
-		batch.GroupName,
-		schema.GroupVersion{Group: batch.GroupName, Version: "v1"},
-		batch.SchemeGroupVersion,
-	)
-
-	resourceEncodingConfig.SetVersionEncoding(
-		autoscaling.GroupName,
-		schema.GroupVersion{Group: autoscaling.GroupName, Version: "v1"},
-		autoscaling.SchemeGroupVersion,
-	)
 
 	storageGroupsToEncodingVersion, err := server.StorageSerialization.StorageGroupsToEncodingVersion()
```

Comment:
if so, I very much want the default `StorageSerialization` enumerated in `TestAPIServerDefaults` so we can see changes as they come in and decide if we need to override them